core/commands/config.go

```diff
@@ -164,11 +165,25 @@ included in the output of this command.
 
 		idmap, ok := cfg["Identity"].(map[string]interface{})
 		if !ok {
-			res.SetError(fmt.Errorf("config has no identity"), cmds.ErrNormal)
+			res.SetError(errors.New("config has no identity"), cmds.ErrNormal)
 			return
 		}
 
-		delete(idmap, "PrivKey")
+		privKeyKey := "" // make sure we both find the name of privkey and we delete it
+		for key, _ := range idmap {
```

Comment:
- and `idmap` will be nil if `cfg["Identity"]` doesnt work and there is `cfg["identity"]` instead. 
- As we discussed, this is turning out to be a mess. easier to move the key out.